src/main/java/uk/gov/gchq/palisade/util/FileResourceBuilder.java

```diff
@@ -104,7 +104,26 @@ private static Resource filesystemScheme(final URI uri) {
 
     @Override
     public Resource build(final URI resourceUri) {
-        return filesystemScheme(resourceUri);
+        URI absoluteResourceId = resourceUri;
+        if (!Path.of(resourceUri.getScheme()).isAbsolute()) {
+            File localResourece = new File(resourceUri.getSchemeSpecificPart());
+            String path;
+            try {
+                path = localResourece.getCanonicalPath();
+            } catch (IOException e) {
+                LOGGER.error("Unable to get the Canonical path value", e);
+                path = localResourece.getAbsolutePath();
+            }
+
+            if (localResourece.isDirectory() && !path.endsWith("/")) {
+                path += "/";
+            }
+            absoluteResourceId = UriBuilder.create(resourceUri)
+                    .withoutScheme().withoutAuthority()
+                    .withPath(path)
+                    .withoutQuery().withoutFragment();
+        }
+        return filesystemScheme(absoluteResourceId);
```

Comment:
add a bit of inline javadoc 
e.g. if the file doesnt start/end with a "/"

Reply:
Updated